src/Path/IO.hs

```diff
@@ -289,3 +292,18 @@ dropRoot (Path l) = Path (FP.dropDrive l)
 ignoreDoesNotExist :: MonadIO m => IO () -> m ()
 ignoreDoesNotExist f =
     liftIO $ catch f $ \e -> unless (isDoesNotExistError e) (throwIO e)
+
+withCanonicalizedSystemTempDirectory :: (MonadMask m, MonadIO m)
+    => String            -- ^ Directory name template.
+    -> (FilePath -> m a) -- ^ Callback that can use the canonicalized directory
```

Comment:
I'd actually been thinking this would be a `Path Abs Dir`, to make it fit in with the rest of this module and avoid the need for the `parseAbsDir` in all of the call sites.